WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -139,9 +139,9 @@ private async Task HandleCoinJoinCommandsAsync(ConcurrentDictionary<string, Coin
 		void StartCoinJoinCommand(StartCoinJoinCommand startCommand)
 		{
 			var walletToStart = startCommand.Wallet;
-			if (trackedCoinJoins.TryGetValue(walletToStart.WalletName, out var tracker) && !tracker.IsCompleted)
+			if (trackedCoinJoins.TryGetValue(walletToStart.WalletName, out var tracker))
```

Comment:
This assumes that is a wallet to start is contained in the `trackedCoinJoins` then it is already running. However that's not 100% true because there is a period of time between its finalization and it removal from the `trackedCoinJoins` where the wallets is not being coinjoined. That's why `tracker.IsCompleted` is completed was used. 

Based on what you said it seems this condition was not enough and we should have to consider also the case where it is not even started.

Reply:
Should we let that case happen, it is a can of worm IMO. An outside controller should wait until the cj client finished its operation and then try to start. The CoinJoin manager respects that, auto restart is scheduled only after removal.

Reply:
> Should we let that case happen

We are not let it happen. It just happens. The question could be: do we really care about those edge cases? Is it safe to ignore them? I think it could be safe to ignore, yes.

Reply:
Yes, you are right, the real question is if we should ignore the request to start. A possible risk is that the CoinJoin will not be started - which is a valid risk. 

Let's call the time interval between the tracker finish and tracked disposal - the critical period. For external players like MusicBox, the status is reported by events. Someone can start CoinJoinClient which is tracked by the tracker. The concept is that CoinJoinManager works with the trackers and the tracker manages the Task inside. However technically we could add and start a tracker in critial period, despite the fact that the tracker is not disposed, should we be that eager to do that - if yes why? We should have a good reason because it would break up the concepts of CoinJoinManager working only with trackers. And not talking about the consequence of not waiting for the proper finish, where coins ara marked as used or SmartCoin.CoinJoinInProgress set back to false, or replacing a living tracker thus forgetting the reference of it (btw would it be disposed later after AddOrUpdate ).  I don't think we should allow this. If the tracker is still there  you cannot start the client again.

Reply:
Basically we are on the same page - merging.